### PR TITLE
ensure macro expansion uses the right module

### LINKED ISSUE #15838: Extending macro in another module
Macro expansion happens in the wrong module if it is extended in another module.

```jl
julia> module A
       macro f()
       end
       const x = :a
       end
A

julia> module B
       import A.@f
       macro f(x)
           :x
       end
       const x = :b
       end
B

julia> import A.@f

julia> @f 1
:a
```

The macro called is defined in `B` so it should expend to `B.x == :b` instead.

### PR DESCRIPTION
**Pull request by vtjnash**
fix #15838

### CODE REVIEW

**Comment by vtjnash** [comment]
CI passed, except for a networking failure in the pkg test. Is this testing something unreliable?

```julia
	From worker 3:	     * pkg                  Exception running test pkg : 
On worker 3:
LoadError: GitError(Code:ERROR, Class:OS, Failed to receive response: The server returned an invalid or unrecognized response
) 
 in macro expansion at .\libgit2\error.jl:98 [inlined] 
 in #fetch#46 at .\libgit2\remote.jl:70 
 in #fetch at .\null:0 
 in #fetch#87 at .\libgit2.jl:159 
 in #18 at .\pkg\entry.jl:234 
 in transact at .\libgit2.jl:499 
 in with at .\libgit2\types.jl:660 
 in checkout at .\pkg\entry.jl:227
 in #2 at .\pkg\dir.jl:31
 in cd at .\file.jl:48
 in #cd#1 at .\pkg\dir.jl:31
 in #checkout#1 at .\pkg.jl:164 
 in #3 at C:\projects\julia\test\pkg.jl:74 
 in temp_pkg_dir at C:\projects\julia\test\pkg.jl:13 
 in temp_pkg_dir at C:\projects\julia\test\pkg.jl:7 
 in include_string at .\loading.jl:380 
 in include_from_node1 at .\loading.jl:429 
 in macro expansion at .\util.jl:179 [inlined] 
 in runtests at C:\projects\julia\test\testdefs.jl:7 
 in #16 at C:\projects\julia\test\runtests.jl:36 
 in #302 at .\multi.jl:1026 
 in run_work_thunk at .\multi.jl:750 
 in macro expansion at .\multi.jl:1026 [inlined] 
 in #301 at .\event.jl:46
while loading C:\projects\julia\test\pkg.jl, in expression starting on line 22 
ERROR: LoadError: Some tests exited with errors.
 in (::##11#19)() at C:\projects\julia\test\runtests.jl:64
 in cd(::##11#19, ::String) at .\file.jl:48
 in include_from_node1(::String) at .\loading.jl:426
 in process_options(::Base.JLOptions) at .\client.jl:266
 in _start() at .\client.jl:322
while loading C:\projects\julia\test\runtests.jl, in expression starting on line 13
Command exited with code 1
```

**Comment by tkelman** [comment]
That error is happening often. Not sure if it's a new test or hitting recently refactored code paths. cc @wildart any ideas?

File: src/cgutils.cpp [hunk 1]
```diff
@@ -1702,7 +1702,7 @@ static jl_cgval_t emit_new_struct(jl_value_t *ty, size_t nargs, jl_value_t **arg
                     if (!init_as_value) {
                         // avoid unboxing the argument explicitely
                         // and use memcpy instead
-                        dest = builder.CreateConstInBoundsGEP2_32(lt, strct, 0, i);
+                        dest = builder.CreateConstInBoundsGEP2_32(LLVM37_param(lt) strct, 0, i);
```
Comment (tkelman) [inline src/cgutils.cpp]: this should perhaps have been mentioned in the commit message, but thanks

**Comment by wildart** [comment]
We'll never know what it was because libgit2 does not report actual error for windows connection.

**Pull request merged**